flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/enumerator/KinesisStreamsSourceEnumeratorStateSerializer.java

```diff
@@ -109,13 +110,22 @@ public KinesisStreamsSourceEnumeratorState deserialize(
                                 + ". Serializer version is "
                                 + splitSerializer.getVersion());
             }
-            Set<KinesisShardSplit> unassignedSplits = new HashSet<>(numUnassignedSplits);
+            List<KinesisShardSplitWithAssignmentStatus> unassignedSplits =
+                    new ArrayList<>(numUnassignedSplits);
             for (int i = 0; i < numUnassignedSplits; i++) {
                 int serializedLength = in.readInt();
                 byte[] serializedSplit = new byte[serializedLength];
                 if (in.read(serializedSplit) != -1) {
+                    KinesisShardSplit deserializedSplit =
+                            splitSerializer.deserialize(splitSerializerVersion, serializedSplit);
+                    SplitAssignmentStatus assignmentStatus = SplitAssignmentStatus.UNASSIGNED;
+                    if (version == CURRENT_VERSION) {
+                        assignmentStatus = SplitAssignmentStatus.fromStatusCode(in.readInt());
+                    }
```

Comment:
for previous serializer version, we should be safe to assume unassigned split right?